sno/repo_files.py

```diff
@@ -0,0 +1,87 @@
+import os
+
+from pathlib import Path
+import shlex
+import subprocess
+
+
+from . import is_windows
+from .exceptions import SubprocessError
+
+
+HEAD = "HEAD"
+COMMIT_EDITMSG = "COMMIT_EDITMSG"
+ORIG_HEAD = "ORIG_HEAD"
+MERGE_HEAD = "MERGE_HEAD"
+MERGE_MSG = "MERGE_MSG"
+MERGE_INDEX = "MERGE_INDEX"
+MERGE_LABELS = "MERGE_LABELS"
+
+
+def repo_file_path(repo, filename):
+    return Path(repo.path) / filename
+
+
+def repo_file_exists(repo, filename):
+    return repo_file_path(repo, filename).exists()
+
+
+def write_repo_file(repo, filename, contents):
+    if not isinstance(contents, str):
+        raise TypeError("File contents must be a string", type(contents))
+    if contents[-1] != "\n":
+        contents += "\n"
+
+    path = repo_file_path(repo, filename)
+    path.write_text(contents, encoding="utf-8")
+
+
+if is_windows:
+    FALLBACK_EDITOR = "notepad.exe"
+else:
+    FALLBACK_EDITOR = "nano"
```

Comment:
```python
    import shutil
    FALLBACK_EDITOR = shutil.which("nano") or "vi"
```

I'd use nano if possible but not all linux distros have it. `vi` is part of POSIX so we should probably use that as a last-ditch fallback if `nano` doesn't exist?

Hopefully users can figure out how to exit it...

Reply:
`nano` is included with default Debian, Centos, Fedora, and Ubuntu.

Reply:
Done - rcoup@ says everyone has nano, but since it is easy to workaround if they don't, fallback-of-fallback to vi is done.